vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -347,6 +367,82 @@ void AdsImpl::SetConfirmationsIsReady(const bool is_ready) {
   is_confirmations_ready_ = is_ready;
 }
 
+std::map<std::string, std::vector<AdsHistory>> AdsImpl::GetAdsHistory() {
+  std::map<std::string, std::vector<AdsHistory>> ads_history;
+  base::Time now = base::Time::Now().LocalMidnight();
+
+  auto ad_history_details = client_->GetAdsShownHistory();
+  for (auto& detail_item : ad_history_details) {
+    // FIXME(erogul): This should probably be done in the JavaScript
+    // WebUI
+    if (detail_item.ad_content.brand_info.size() > 50) {
+      detail_item.ad_content.brand_info.resize(50);
+      detail_item.ad_content.brand_info += "...";
+    }
+
+    auto history_item = std::make_unique<AdsHistory>();
+    history_item->details.push_back(detail_item);
+
+    base::Time timestamp = base::Time::FromDeltaSinceWindowsEpoch(
+        base::TimeDelta::FromSeconds(detail_item.timestamp_in_seconds));
+    base::TimeDelta time_delta = now - timestamp.LocalMidnight();
+    if (time_delta.InDays() >= kDaysOfAdsHistory) {
+      break;
+    }
+
+    std::string timestamp_numeric =
+        base::UTF16ToUTF8(base::TimeFormatShortDateNumeric(timestamp));
+    ads_history[timestamp_numeric].push_back(*history_item);
+  }
+
+  return ads_history;
+}
+
+int AdsImpl::ToggleAdThumbUp(const std::string& id,
+                             const std::string& creative_set_id,
+                             int action) {
+  int like_action = client_->ToggleAdThumbUp(id, creative_set_id, action);
+
+  if (like_action == 0) {
+  } else {
+    // FIXME(erogul): Send a confirmation
+  }
```

Comment:
@tmancey Could use some advice on how to make this happen, as it requires a `NotificationInfo` which I don't have access to here.

Reply:
@tmancey I ended up creating a `NotificationInfo` object here and filling in the two fields that are used. Probably not ideal, but I didn't want to thread `RedeemToken` through all of the interfaces (although I guess another approach would be a version of `ConfirmAd` that only takes the two required params...)

Reply:
I would go with another function named something like ConfirmAction which only takes the params it needs

Reply:
Makes sense, will do.